test/base/test_batch.py

```diff
@@ -837,10 +859,11 @@ def test_slice_distribution() -> None:
         selected_idx = [1, 3]
         sliced_batch = batch[selected_idx]
         sliced_probs = cat_probs[selected_idx]
-        assert (sliced_batch.dist.probs == Categorical(probs=sliced_probs).probs).all()
-        assert (
-            Categorical(probs=sliced_probs).probs == get_sliced_dist(dist, selected_idx).probs
-        ).all()
+        assert torch.allclose(sliced_batch.dist.probs, Categorical(probs=sliced_probs).probs)
```

Comment:
Thx! I guess it was failing randomly, right?

Reply:
Yes, it was sporadically returning inequality. I will take a look at __getitem__ soon as I have an overdue task there with maintaining slicing consistency. Then I can also check what causes slicing to make these little decimal errors.

Reply:
Thanks, appreciate it! I don't think the getitem issue can actually be solved, but implementing eq already solves all practical problems of it. So no need to further address it, unless you see some simple way for doing that. 

The order of slicing however it's still a valid issue